src/firefly/js/ui/UploadTableSelectorSingleCol.jsx

```diff
@@ -0,0 +1,231 @@
+ * from the uploaded file.
+ *
+ * As of 7/12/2024, ObjectIDSearch on TAP and Search by ID in Euclid make use of this.
+ *
+ * If you just want a regular UploadTableSelector that'll have 2 Position Columns (ra and dec) from the uploadd file,
+ * please use the UploadTableSelectorPosCol component
+ * @param props
+ * @param props.uploadInfo
+ * @param props.setUploadInfo
+ * @param props.headerTitle
+ * @param props.colName
+ * @param props.setSelectInObjList used for ObjectID on TAP
+ * @param props.getUseSelectIn used for ObjectID on TAP
+ * @param props.setWorking used for ObjectID on TAP
+ * @returns {JSX.Element}
+ */
+export function UploadTableSelectorSingleCol({uploadInfo, setUploadInfo,
+                                                 headerTitle='Uploaded Column:',
+                                                 colName = '',
+                                                 setSelectInObjList = () => {}, getUseSelectIn = () => {}, setWorking = () => {} }) {
+    const [getSingleCol,setSingleCol]= useFieldGroupValue(UploadSingleColumn);
+    const {fileName,columns,totalRows,fileSize}= uploadInfo ?? {};
+    const columnsUsed= columns?.filter( ({use}) => use)?.length ?? 0;
+    const openKey= 'upload-single-column';
+
+    useEffect(() => {
+        //if user changes position column(s), make the new columns entries selectable in the columns/search
+        const columns = uploadInfo?.columns;
+        if (columns?.length===1) {
+            setSingleCol(columns[0].name);
+        }
+        if (getSingleCol()) {
+            const cObj= columns.find((col) => col.name === getSingleCol());
+            if (cObj) cObj.use = true;
+        }
+        uploadInfo = {...uploadInfo, columns};
+        setUploadInfo(uploadInfo);
+        if (getUseSelectIn()==='use') {
+            loadTableColumn(getSingleCol(),uploadInfo.serverFile,setSelectInObjList,setWorking);
+        }
+    }, [getSingleCol, getUseSelectIn]);
+
+    const preSetUploadInfo= (ui) => {
+        setSingleCol('', {validator: getColValidator(ui.columns, true, false), valid: true});
+        setUploadInfo(ui);
+    };
+
+    const haveFile= Boolean(fileName && columns);
+
+    const onColsSelected = (selectedColNames) => {
+        //get rid of extra quotes within each selectedColNames - because non-alphanumeric entries may have
+        //been quoted by calling quoteNonAlphanumeric
+        // , e.g.: ['"Object Name"', 'RA', 'Notes']
+        selectedColNames = selectedColNames.map((col) => col.replace(/^"(.*)"$/, '$1'));
+        const columns = uploadInfo?.columns.map((col) => (
+            {...col, use:selectedColNames.includes((col.name))}));
+        uploadInfo = {...uploadInfo, columns};
+        setUploadInfo(uploadInfo);
+    };
+
+    return (
+        <div style={{margin: '10px 0 0 0'}}>
+            <div style={{display:'flex', alignItems:'center'}}>
+                <div style={{display:'flex', alignItems:'center'}}>
```

Comment:
Why are you not using JoyUI in new code?

Reply:
this part was copied over from an existing component in `ObjectIDSearch.jsx`, but yes, I'll go ahead and make this JoyUI!

Reply:
Are you still planning to convert this part? I think you should